drivers/gpio/gpio_xlnx_axi.c

```diff
@@ -193,38 +210,151 @@ static int gpio_xlnx_axi_port_toggle_bits(const struct device *dev,
 	return 0;
 }
 
-static int gpio_xlnx_axi_pin_interrupt_configure(const struct device *dev,
-						 gpio_pin_t pin,
-						 enum gpio_int_mode mode,
-						 enum gpio_int_trig trig)
+/**
+ * Enables interrupts for the given pins on the channel
+ * The axi gpio can only enable interrupts for an entire port, so we need to track
+ * the pins and modes ourselves.
+ */
+static int gpio_xlnx_axi_pin_interrupt_configure(const struct device *dev, gpio_pin_t pin,
+						 enum gpio_int_mode mode, enum gpio_int_trig trig)
 {
-	ARG_UNUSED(dev);
-	ARG_UNUSED(pin);
-	ARG_UNUSED(mode);
-	ARG_UNUSED(trig);
```

Comment:
You need to preserve these in the `#else` case.

Reply:
Re-added.